packages/frontend/src/hooks/usePositions.ts

```diff
@@ -50,54 +59,75 @@ export const usePnL = () => {
+        const result = await calcETHCollateralPnl(vaultHistory, toTokenAmount(index, 18).sqrt(), existingCollat)
+        setEthCollateralPnl(result)
+      }
+    })()
+  }, [index.toString(), existingCollat.toString(), vaultHistory?.length])
 
   useEffect(() => {
     if (!ready || positionLoading) return
 
     const p1 = getSellQuote(squeethAmount).then(setSellQuote)
     const p2 = getBuyQuote(squeethAmount).then((val) => setBuyQuote(val.amountIn))
     Promise.all([p1, p2]).then(() => setLoading(false))
-  }, [ready, squeethAmount.toString(), shortVaults?.length, positionLoading])
+  }, [getBuyQuote, getSellQuote, positionLoading, ready, squeethAmount.toString()])
 
   useEffect(() => {
     if (sellQuote.amountOut.isZero() && !loading) {
       setLongGain(new BigNumber(0))
       return
     }
-    const _currentValue = sellQuote.amountOut.times(100).div(wethAmount.absoluteValue())
-    const _gain = _currentValue.minus(100)
+    // const _currentValue = sellQuote.amountOut.times(100).div(wethAmount.absoluteValue())
+    const _gain = longUnrealizedPNL.usd.dividedBy(totalUSDFromBuy).times(100)
     setLongGain(_gain)
-  }, [ethPrice.toString(), wethAmount.toString(), sellQuote.amountOut.toString(), squeethAmount.toString()])
-
-  const shortUnrealizedPNL = useMemo(
-    () => calcUnrealizedPnl({ wethAmount, buyQuote, ethPrice }),
-    [buyQuote.toString(), ethPrice.toString(), wethAmount.toString()],
-  )
-
-  const longUnrealizedPNL = useMemo(
-    () => calcLongUnrealizedPnl({ sellQuote: sellQuote.amountOut, wethAmount, ethPrice }),
-    [ethPrice.toString(), sellQuote.amountOut.toString(), wethAmount.toString()],
-  )
+  }, [loading, longUnrealizedPNL.usd.toString(), sellQuote.amountOut.toString(), totalUSDFromBuy])
 
   useEffect(() => {
-    if (squeethAmount.isZero()) {
+    if (squeethAmount.isZero() || shortUnrealizedPNL.usd.isZero()) {
       setLongGain(new BigNumber(0))
       return
     }
-    const _gain = calcShortGain({ shortUnrealizedPNL, usdAmount: shortUsdAmount, wethAmount, ethPrice })
+    const _gain = shortUnrealizedPNL.usd
+      .dividedBy(totalUSDFromSell.plus(existingCollat.times(toTokenAmount(index, 18).sqrt())))
+      .times(100)
+
     setShortGain(_gain)
   }, [
-    ethPrice.toString(),
-    shortUnrealizedPNL.toString(),
+    index.toString(),
+    existingCollat.toString(),
+    shortUnrealizedPNL.usd.toString(),
     squeethAmount.toString(),
-    shortUsdAmount.toString(),
-    wethAmount.toString(),
+    totalUSDFromSell.toString(),
   ])
 
+  useEffect(() => {
+    ;(async () => {
+      if (swaps?.length && !buyQuote.isZero() && !index.isZero() && !ethCollateralPnl.isZero()) {
+        const pnl = await calcDollarShortUnrealizedpnl(swaps, isWethToken0, buyQuote, toTokenAmount(index, 18).sqrt())
+        setShortUnrealizedPNL({
+          usd: pnl.plus(ethCollateralPnl),
+          eth: pnl.plus(ethCollateralPnl).div(toTokenAmount(index, 18).sqrt()).isFinite()
```

Comment:
Just wondering if we take ethPrice from ethDai pool instead? Or it will be totally equal to square root of index, not sure if there will be some slight differences between two numbers

Reply:
Cc @alexisgauba

Reply:
We are using the ETH-USDC pool. Index = eth-usdc price squared so should be the same as taking the square root